server/chat-commands.js

```diff
@@ -621,20 +621,33 @@ const commands = {
 	'!away': true,
 	idle: 'away',
 	afk: 'away',
-	away(target, room, user) {
-		this.parse('/blockchallenges');
-		this.parse(`/blockpms ${target}`);
+	brb: 'away',
+	away(target, room, user, connection, cmd) {
+		let awayMessage;
+		if (target) {
+			awayMessage = Chat.namefilter(target, user);
+			if (!awayMessage) return;
+		} else {
+			awayMessage = toId(cmd);
+			if (awayMessage === 'afk' || awayMessage === 'brb') {
+				awayMessage = awayMessage.toUpperCase();
+			} else {
+				awayMessage = `${awayMessage[0].toUpperCase()}${awayMessage.slice(1)}`;
+			}
+		}
+		user.setAway(awayMessage);
+		this.sendReply("You are now marked as away. Send a message or use /back to indicate you are back.");
 	},
-	awayhelp: [`/away - Blocks challenges and private messages. Unblock them with /back.`],
+	awayhelp: [`/away - Marks you as away. Send a message or use /back to indicate you are back.`],
 
 	'!back': true,
 	unaway: 'back',
 	unafk: 'back',
-	back() {
-		this.parse('/unblockpms');
-		this.parse('/unblockchallenges');
+	back(target, room, user) {
```

Comment:
not sure you're using `target, room, user` here?

Reply:
user.setBack

Reply:
right, my bad!